src/Illuminate/Foundation/Application.php

```diff
@@ -1105,14 +1105,24 @@ public function addAbsoluteCachePathPrefix($prefix)
         return $this;
     }
 
+    /**
+     * Get the instance of the MaintenanceMode.
+     *
+     * @return \Illuminate\Contracts\Foundation\MaintenanceMode
+     */
+    public function maintenanceMode()
+    {
+        return $this->make(MaintenanceMode::class);
```

Comment:
Now that you have a contract, it's probably better to resolve that. Otherwise you won't be able to overwrite the concrete implementation. We'll need to bind the contract to the container.

Reply:
Check. I've now bind the contract to the default implementation in the src/Illuminate/Foundation/Providers/FoundationServiceProvider.php. I was not totally sure if thats the correct place to do it.